test/test_hexhamming.py

```diff
@@ -185,3 +245,43 @@ def test_hamming_distance_bytes_bench(benchmark, hex1, hex2):
 
 def test_check_hexstrings_within_dist_bench(benchmark):
     benchmark(check_hexstrings_within_dist, "F" * 1000, "0" * 1000, 20)
+
+
+@pytest.mark.benchmark(group="hamming_distance_bytes_arrays_within_dist")
+@pytest.mark.parametrize(
+    ("bytes1", "bytes2", "max_dist", "result"),
+    (
+        (b"\x00" * 16 + b"\x00\x03" * 8 * 511,
+         b"\x00" * 16, 1, 0),
+        (b"\x00\x03" * 8 * 256 + b"\x00" * 16 + b"\x00\x03" * 8 * 255,
+         b"\x00" * 16, 1, 256),
+        (b"\x00\x03" * 8 * 511 + b"\x00" * 16,
+         b"\x00" * 16, 1, 511),
+        (b"\xFF" * 32 + b"\x11" * 32 * 1023,
+         b"\xFB" * 32, 4*32, 0),
+        (b"\x11" * 32 * 511 + b"\xFF" * 32 + b"\x11" * 32 * 512,
+         b"\xFB" * 32, 4*32, 511),
+        (b"\x11" * 32 * 1023 + b"\xFF" * 32,
+         b"\xFB" * 32, 4*32, 1023),
+        (b"\xCC" * 64 + b"\x01" * 64 * 16383,
+         b"\xFB" * 64, 5*64, 0),
+        (b"\x01" * 64 * 8191 + b"\xCC" * 64 + b"\x01" * 64 * 8192,
+         b"\xFB" * 64, 5*64, 8191),
+        (b"\x01" * 64 * 16383 + b"\xCC" * 64,
+         b"\xFB" * 64, 5*64, 16383),
+    ),
+    ids=(
+        "  512 elems,s=16,at 0",
+        "  512 elems,s=16,mid",
+        "  512 elems,s=16,end",
+        " 1024 elems,s=32,at 0",
+        " 1024 elems,s=32,mid",
+        " 1024 elems,s=32,end",
+        "16384 elems,s=64,at 0",
+        "16384 elems,s=64,mid",
+        "16384 elems,s=64,end",
+    ),
+)
+def test_check_bytes_arrays_within_dist_bench(benchmark, bytes1, bytes2, max_dist, result):
+    result = benchmark(check_bytes_arrays_within_dist, bytes1, bytes2, max_dist)
+    assert result == result
```

Comment:
You can either omit this `assert` or rename the passed in `result` to `expected`. I'm happy for you to just omit the `assert` given that you're already testing `check_bytes_arrays_within_dist`.